<launch>
	<node name="localization" pkg="handle_detector" type="handle_detector_localization" output="screen">
		<!-- affordance search parameters -->
    <param name="file" value="" />
		<param name="target_radius" value="0.012" />
		<param name="target_radius_error" value="0.012" />
		<param name="affordance_gap" value="0.008" />
		<param name="sample_size" value="1000" />		
		<param name="use_clearance_filter" value="true" />
		<param name="use_occlusion_filter" value="true" />
    <param name="curvature_estimator" value="1" />
		<param name="point_cloud_source" value="1" />
		<param name="update_interval" value="30.0" />
		
		<!-- alignment parameters -->
		<param name="alignment_runs" value="10" />
		<param name="alignment_min_inliers" value="8" />
		<param name="alignment_dist_radius" value="0.03" />
		<param name="alignment_orient_radius" value="0.4" />
		<param name="alignment_radius_radius" value="0.003" />
				
		<!-- workspace limits -->
		<param name="max_range" value="0.9" />
   		<param name="workspace_min_x" value="-1.4" />
		<param name="workspace_max_x" value="2.5" />
		<param name="workspace_min_y" value="-0.9" />
		<param name="workspace_max_y" value="0.9" />
		<param name="workspace_min_z" value="-1.2" />
		<param name="workspace_max_z" value="1.2" />
		
		<!-- number of threads to use -->
		<param name="num_threads" value="4" />
	</node>

    <node pkg="handle_tracking" type="scanner" name="handle_tracker_node" output="screen"/>
    <node pkg="villa_manipulation" type="handover_activity.py" name="opendoor_server" output="screen"/>
    <node pkg="villa_navi_service" type="waypoint_server" name="waypoint_server" output="screen" respawn="true"></node>

    <arg name="param_file" default="$(find villa_navi_service)/config/naviconfig.yaml" />
    <rosparam command="load" file="$(arg param_file)"/>
    <arg name="target_frame" default="/map" />

</launch>
